### DISCUSSION TITLE
Making RavenTestDriver faster

### DISCUSSION THREAD

**Discussion by iainxt** [ravendb/ravendb#14209] [Q&A]
I am using RavenTestDriver and I have noticed it takes around 16 seconds to run unit tests. 

Calling `GetDocumentStore` takes 8 seconds. Calling `ExecuteIndex` takes 8 seconds.

Is this the expected speed? Is there anyway to speed this up?

I tried to make it In Memory but I cannot find the setting.

```
    public class RavenDBTestDriver3 : RavenTestDriver
    {
        [Fact]
        public void MyFirstTest()
        {
            var start = DateTime.UtcNow;
            ConfigureServer(new TestServerOptions
            {
            });
            TimeSpan configureServer = DateTime.UtcNow - start;
            start = DateTime.UtcNow;
            var id = Guid.NewGuid().ToString();
            using var store = GetDocumentStore();
            var getStore = DateTime.UtcNow - start;
            start = DateTime.UtcNow;
            store.ExecuteIndex(new TestDocumentByName());
            var addIndex = DateTime.UtcNow - start;
            start = DateTime.UtcNow;
        }
    }
```

**Comment by ayende** [comment]
It is already running in memory.
Note that what we do is spawn a slaved process to your unit test. The first time you call it, may be a bit slow.
Taking 8 seconds is beyond anything that I saw. Do you have some filtering / Anti Virus that may be involved?

**Reply by iainxt** [reply]
I might have some anti virus involved, I'll try troubleshooting via a new project and also on a different laptop.

I saw some reference else where to including the licence file to get multiple cores. Could this make the setup faster?

**Reply by ppekrol** [reply]
Licensing is being applied few seconds after server starts, it would help with processing during the tests, but it looks like you have issues with server startup time? First call to 'GetDocumentStore' starts the Embedded server and keeps the process running for the whole test suite.

BTW, I've noticed that you are using 'ConfigureServer' inside a test. This is a one-time operation, you want to do it once in lifetime of your application, in xunit a static ctor in the test base is usually a good option.
